### PR TITLE
Converted FeatureBox to a link and added hover effect

### PR DESCRIPTION
**Pull request by patrykbn**
FeatureBox.js:
Converted content of FeatureBox to be a clickable link. Link temporarily disabled so that is does not navigate to a different page.
Additionaly added a PropType to check the content of link and require a string.

FeatureBox.module.scss:
added styles for added link with !important to overwrite default underline on hover
added transition time and style for hover effect to: .root .iconWrapper and .content
added hover effect styles to match active state for the FeatureBox
deleted empty style for p that was causing a warning

FeatureBoxes.js:
added link prop. to FeatureBox with a placeholder for future implementation

JIRA ticket: 
https://projects.kodilla.com/browse/WDP240901-3

### CODE REVIEW

**Review by dave8git — DISMISSED** [review]

File: src/components/common/FeatureBox/FeatureBox.js [hunk 1]
```diff
@@ -5,21 +5,28 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
...
-const FeatureBox = ({ active, icon, children }) => (
+const FeatureBox = ({ active, icon, children, link }) => (
   <div className={styles.root + (active ? ' ' + styles.active : '')}>
-    {icon && (
-      <div className={styles.iconWrapper}>
-        <FontAwesomeIcon className={styles.icon} icon={icon} />
-      </div>
-    )}
-    <div className={styles.content}>{children}</div>
+    <a
```
Comment (matwjablonski) [inline src/components/common/FeatureBox/FeatureBox.js:10]: it should not be an anchor element, it should be Link component from react-router library

File: src/components/common/FeatureBox/FeatureBox.module.scss [hunk 2]
```diff
@@ -75,4 +79,20 @@
       color: $primary;
     }
   }
+
+  // Hover effect
+  &:hover {
+    .iconWrapper {
+      color: #ffffff;
```
Comment (matwjablonski) [inline src/components/common/FeatureBox/FeatureBox.module.scss:86]: this color should be taken from variables

**Review by matwjablonski — CHANGES_REQUESTED** [review]

File: src/components/common/FeatureBox/FeatureBox.module.scss (hunk 2 above)
Comment (patrykbn) [reply src/components/common/FeatureBox/FeatureBox.module.scss:86]: This was done before the variables were added, but its true. I'll change it

**Review by dave8git — DISMISSED** [review]

File: src/components/common/FeatureBox/FeatureBox.module.scss (hunk 2 above)
Comment (matwjablonski) [reply src/components/common/FeatureBox/FeatureBox.module.scss]: colors are merged to main branch :) now you can align this part ;)

**Review by dave8git — APPROVED** [review]

**Review by matwjablonski — APPROVED** [review]

**Pull request merged**